tests/integration/suite/test_workloads.py

```diff
@@ -310,6 +311,34 @@ def test_statefulset_workload_volumemount_subpath(admin_pc):
         assert e.value.error.status == 422
 
 
+def test_workload_redeploy(admin_pc, admin_mc):
+    project = admin_mc.client.create_project(name=random_str(),
+                                             clusterId='local')
+    ns = admin_pc.cluster.client.create_namespace(
+        name=random_str(),
+        projectId=project.id)
+    p_client = user_project_client(admin_mc, project)
+    name = random_str()
+    workload = p_client.create_workload(
+        name=name,
+        namespaceId=ns.id,
+        scale=1,
+        containers=[{
+            'name': 'one',
+            'image': 'nginx',
+        }],
+        annotations={
+            "cattle.io/timestamp":
+                datetime.datetime.now().isoformat()}, )
+    p_client.action(workload, "redeploy")
+    timestamp = p_client.list_workload(
+        uuid=workload.uuid).data[0].annotations["cattle.io/timestamp"]
+    assert timestamp != workload.annotations["cattle.io/timestamp"]
+    p_client.delete(workload)
+    admin_pc.cluster.client.delete(ns)
+    admin_mc.client.delete(project)
```

Comment:
The admin_pc fixture handles the project creation/deletion so we don't need to do that in the tests